src/ppx/metrics.cpp

```diff
@@ -255,7 +324,15 @@ bool Manager::HasActiveRun() const
     return (mActiveRun != nullptr);
 }
 
-MetricID Manager::AddMetric(const MetricMetadata& metadata)
+MetricID Manager::EnsureAllocateID(MetricID reuseID)
```

Comment:
This doesn't ensure the ID is allocated, it only ensures the ID is not kInvalidMetricID.
- doesn't prevent collisions
- doesn't actually check if the number is allocated, or even the possible range `[0; mNextMetricID[`